src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -263,6 +263,9 @@ protected function resolveExprClassConstFetch(Expr\ClassConstFetch $node)
         if ('class' === $constantName) {
             return $refClass->getName();
         }
+        
+        $this->isConstant = true;
+        $this->constantName = (string)$node->class . '::' . $constantName;
```

Comment:
What if this isn't constant from same class (e.g. `self::CONST_NAME`), but constant from other class (e.g. `OtherClass::CONST_NAME`) or even namespaced constant (declared via `const ...`)?

Would this work and is this covered in test case?